extensions/amp-document-recommendations/0.1/amp-document-recommendations.js

```diff
@@ -203,7 +203,7 @@ export class AmpDocumentRecommendations extends AMP.BaseElement {
               'failed to parse content discovery script', error);
         });
 
-    this.config_ = assertConfig(configJson);
+    this.config_ = assertConfig(configJson, this.win.document.location.host);
```

Comment:
host won't be correct in the case where document is served from AMP cache. What you need is 
```
const docInfo = Services.documentInfoForDoc(this.element);
const host = parseUrl(docInfo.sourceUrl).host;
```
instead of `this.win.document.location.host`